### PR TITLE
chores: fix the unit test for notebook-controller

### PR DESCRIPTION
**Pull request by harshad16**
chores: fix the unit test for notebook-controller 

The module sigs.k8s.io/controller-runtime/tools/setup-envtest@latest  has 
https://github.com/kubernetes-sigs/controller-runtime/blob/cc2a25b511876693d61c8ad07936824647a8f233/tools/setup-envtest/go.mod#L20
Which requires the base go lang version to be updated.
To fix, this currently, we reverting it to an older version
based on: https://pkg.go.dev/sigs.k8s.io/controller-runtime/tools/setup-envtest?tab=versions

Fixes: https://github.com/opendatahub-io/kubeflow/issues/203

### CODE REVIEW

**Comment by shalberd** [comment]
the Dockerfile, ah :-) cool cool

I had been looking around at https://steps.ci.openshift.org/job?org=opendatahub-io&repo=kubeflow&branch=v1.7-branch&test=kf-notebook-controller-unit to figure out more about why the unit tests themselves even require golang 1.19, but that is a different story, my lack of understanding.

**Comment by harshad16** [comment]
> the Dockerfile, ah :-) cool cool

Still exploring, i was running few checks to see if this work.


> 
> I had been looking around at https://steps.ci.openshift.org/job?org=opendatahub-io&repo=kubeflow&branch=v1.7-branch&test=kf-notebook-controller-unit to figure out more about why the unit tests themselves even require golang 1.19, but that is a different story, my lack of understanding. I mean, is it a must to force go 1.19 on kubeflow here, could 
we not do something at the level where those units tests are specified?

yes , we can probably add the version on the level
However just wanted to get the idea, what is happening here, before making that change.

> 
> https://prow.ci.openshift.org/prowjob?prowjob=8a024592-45e3-4114-9e36-b48fc804fe0e
> 
> https://prow.ci.openshift.org/?repo=opendatahub-io%2Fkubeflow
> 
> All PRs are running across this error, for example Bartosz's work on service mesh overlay.
> 
> Is it correct that out test infra code is based on [upstream kubernetes/test-infra](https://github.com/kubernetes/test-infra/blob/fded157aa8fbffda01e621c025ce6c341dee1fe8/hack/tools/go.mod#L271)?
> 
> I mean, for example the go 1.17 requirement on notebook controller unit test from github workflow is clearly defined
> 
> https://github.com/opendatahub-io/kubeflow/blob/v1.7-branch/.github/workflows/notebook_controller_unit_test.yaml

Yes it is based on openshift/release which is based on test-infra

**Comment by bartoszmajsak** [comment]
This will unblock #172 :)

**Comment by shalberd** [comment]
@bartoszmajsak correct, we want to make those unit tests work again :-)

File: components/notebook-controller/config/rbac/role.yaml [hunk 1]
```diff
@@ -29,7 +29,6 @@ rules:
   - get
   - list
   - watch
-  - delete
```
Comment (shalberd) [inline components/notebook-controller/config/rbac/role.yaml:32]: why this change to not allow deleting pods?

Comment (harshad16) [reply components/notebook-controller/config/rbac/role.yaml:32]: sorry , this just WIP
it would be added back ,  updating it

**Comment by shalberd** [comment]
meaning we can now set back to go.mod go version requirement back to 1.17 / or in line with upstream, correct?

**Comment by harshad16** [comment]
Merging as this fixes the unit test for all the use-cases.
Thanks all for the comments.
